### PR TITLE
generic position controller status/feedback message

### PR DESCRIPTION
**Pull request by dagar**
Not done, but should give you an idea.

TODO
 - yaw acceptance (+ error threshold?)
 - z acceptance could come from the position controller
 - eliminate as much vehicle specific code from navigator as possible
 - consider vtol usage for back transition
 - stopping distance (loiter/rtl/geofence braking)

### CODE REVIEW

**Comment by dagar** [comment]
TODO: vtol implications (multi topic).

**Comment by dagar** [comment]
Ok I think this is everything we need right now. Needs testing and review.

File: msg/position_controller_status.msg [hunk 1]
```diff
@@ -0,0 +1,14 @@
...
+float32 nav_bearing
+
+float32 target_bearing
+float32 xtrack_error
+
+float32 wp_dist
+
+float32 acceptance_radius		# the optimal distance to a waypoint to switch to the next
+
+float32 yaw_acceptance			# NaN if not set
```
Comment (mrivi) [inline msg/position_controller_status.msg]: `yaw_acceptance` should be feedback into navigator, right? From #10240,  dc99bb088ad9c1a1b28a0db22e00a896bc9355d6

**Comment by mrivi** [comment]
@dagar can we get this PR tested?

File: msg/position_controller_status.msg (hunk 1 above)
Comment (dagar) [reply msg/position_controller_status.msg]: Yes.

**Comment by dagar** [comment]
@PX4/testflights could you test this branch on FW or VTOL (must transition to FW)? The test should include at least a couple waypoints.

**Comment by dannyfpv** [comment]
pr tested on v3 fw
https://review.px4.io/plot_app?log=851676a3-f71a-419f-b253-2d5aa822c350
https://review.px4.io/plot_app?log=54f1fa1b-25c3-405d-b673-3ea058310f2a
https://review.px4.io/plot_app?log=9586fcb4-e87a-4d51-860d-6ca7b709ef45
https://review.px4.io/plot_app?log=aa094d9e-68d5-4c01-bf55-22f6a790d1a0

**Comment by dagar** [comment]
Thanks @dannyfpv 

I did a little more testing in SITL and everything looks good except for a small regression in FW landing abort behaviour that I'll have fixed shortly.

**Comment by mrivi** [comment]
@dagar have you made any progress?

**Comment by dagar** [comment]
@mrivi good to go.

**Comment by mrivi** [comment]
@dagar do we need more tests or can we merge directly?

**Pull request merged**